src/coordinates.rs

```diff
@@ -0,0 +1,144 @@
+//! Custom data types used in the Gerber format.
+
+use std::convert::{From, Into};
+
+
+/// The coordinate format specifies the number of integer and decimal places in
+/// a coordinate number. For example, the `24` format specifies 2 integer and 4
+/// decimal places. The number of decimal places must be 4, 5 or 6. The number
+/// of integer places must be not more than 6. Thus the longest representable
+/// coordinate number is `nnnnnn.nnnnnn`. The same format must be defined for X
+/// and Y. Signs in coordinates are allowed; the `+` sign is optional.
+#[derive(Debug, Copy, Clone)]
+pub struct CoordinateFormat(pub u8, pub u8);
+
+
+/// Decimals are a sequence of one or more digits with an optional decimal
+/// point optionally preceded by a `+` or a `-` sign. They must fit in an IEEE
+/// double.
+///
+/// The value is stored as a 64 bit integer with 9 decimal places.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct Decimal(i64);
```

Comment:
No, they're related: Coordinates are stored internally as `Decimal`s. But when generating Gerber code, they're formatted to a coordinate integer-string depending on the currently active `CoordinateFormat`.